From patchwork Mon Jul  3 16:41:29 2023
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Sebastian Reichel <sebastian.reichel@collabora.com>
X-Patchwork-Id: 13300273
Return-Path: 
 <linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id 36DF9C0015E
	for <linux-rockchip@archiver.kernel.org>;
 Mon,  3 Jul 2023 16:41:48 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:Message-Id:Date:Subject:Cc
	:To:From:Reply-To:Content-ID:Content-Description:Resent-Date:Resent-From:
	Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:In-Reply-To:References:
	List-Owner; bh=D/l/70EI9ID+xMAaNZNnKn0H/Nx0SVMSJ0wBOQfT8G8=; b=D4sDMFTkbXLt5v
	HHizSxrldFUI131vFPYA1Y2uQEO9ZEdKZoyn1o5RB8OQiDgmMkZ/enR+8JNZuUjUgC/ExtrGLJVkS
	QougFB4DKsWgY3X4+jjiiTbJnWsdfzIrtFt0lUh8cQL3vF54/G4/dFtKDXhuz9tlgazDhqcmAAHan
	mMGZLO/hfRI249WWFZS9Sw2wCHpuV7D3BT/vNeenPRTGptZudGO99Zsi7nxjxv602W72I7NqaHXRy
	L6SOgwNixoerz4pDpHjudPb0+HGXRMPgUEcPHZMMvSYnFkrmHZrxxIJT3aNuKSYRdbMg8tKaa8rY3
	c8lILcFjRXdX/f2y4RSA==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.96 #2 (Red Hat Linux))
	id 1qGMcO-00B1nN-0r;
	Mon, 03 Jul 2023 16:41:40 +0000
Received: from madras.collabora.co.uk ([46.235.227.172])
	by bombadil.infradead.org with esmtps (Exim 4.96 #2 (Red Hat Linux))
	id 1qGMcL-00B1m9-20
	for linux-rockchip@lists.infradead.org;
	Mon, 03 Jul 2023 16:41:39 +0000
Received: from jupiter.universe (dyndsl-091-248-215-029.ewe-ip-backbone.de
 [91.248.215.29])
	(using TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits)
	 key-exchange X25519 server-signature RSA-PSS (4096 bits) server-digest
 SHA256)
	(No client certificate requested)
	(Authenticated sender: sre)
	by madras.collabora.co.uk (Postfix) with ESMTPSA id 01E206606F6C;
	Mon,  3 Jul 2023 17:41:34 +0100 (BST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/simple; d=collabora.com;
	s=mail; t=1688402494;
	bh=B0URFXgImNMOKALU8VS0EvBoWTwaU4+U/GaqYEktyFs=;
	h=From:To:Cc:Subject:Date:From;
	b=BMfTbVp3pDQZF4LAVvTLSGGR61z5ggRm8jnjTib/HV9Ot5jRcI/poXVrBbrYueekA
	 WjIMbrhiSEL+r1a4KraEd+kbEiph37/qfUkFTLgeJRAyqy/3tL7mRrN6iIHt3vA7/s
	 YEyM82LNnqqlU2hPD3IF1yAjGLlJTtZgdCPAgT1D2TluR0pjVMc0IBNjoDmXliAz+G
	 onWLTXc2lzkHcu14r9+fYsE6P3EIy38Rx1XFhTepLqu3uvFC6djLi/pjOR0jv5YUlL
	 Hz0BNoZze96xk6jMfsMbZAaTCO4nxYElSF2qXc8xMlwsmI3LHwgfQK0+ZEyo8daPr0
	 AQWYPIxT299tw==
Received: by jupiter.universe (Postfix, from userid 1000)
	id 025444829F6; Mon,  3 Jul 2023 18:41:30 +0200 (CEST)
From: Sebastian Reichel <sebastian.reichel@collabora.com>
To: Marc Zyngier <maz@kernel.org>
Cc: Chris Morgan <macromorgan@hotmail.com>,
	Thomas Gleixner <tglx@linutronix.de>,
	Heiko Stuebner <heiko@sntech.de>,
	linux-rockchip@lists.infradead.org,
	linux-kernel@vger.kernel.org,
	Sebastian Reichel <sebastian.reichel@collabora.com>,
	kernel@collabora.com
Subject: [PATCH v1 1/1] irqchip/gic-v3: Enable Rockchip 3588001 erratum
 workaround for RK3588S
Date: Mon,  3 Jul 2023 18:41:29 +0200
Message-Id: <20230703164129.193991-1-sebastian.reichel@collabora.com>
X-Mailer: git-send-email 2.40.1
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20230703_094137_841876_993120BC 
X-CRM114-Status: GOOD (  12.97  )
X-BeenThere: linux-rockchip@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: Upstream kernel work for Rockchip platforms
 <linux-rockchip.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-rockchip/>
List-Post: <mailto:linux-rockchip@lists.infradead.org>
List-Help: <mailto:linux-rockchip-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=subscribe>
Sender: "Linux-rockchip" <linux-rockchip-bounces@lists.infradead.org>
Errors-To: 
 linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org

Commit a8707f553884 ("irqchip/gic-v3: Add Rockchip 3588001 erratum
workaround") mentioned RK3588S (the slimmed down variant of RK3588)
being affected, but did not check for its compatible value. Thus the
quirk is not applied on RK3588S. Since the GIC ITS node got added to the
upstream DT, boards using RK3588S are no longer booting without this
quirk being applied.

Fixes: 06cdac8e8407 ("arm64: dts: rockchip: add GIC ITS support to rk3588")
Signed-off-by: Sebastian Reichel <sebastian.reichel@collabora.com>
---
I recently got a Rock 5A and noticed this issue. Apart from it, the Indiedroid
Nova should also be affected (I don't have that board). There are no other
upstream RK3588S boards at the moment.
---
 drivers/irqchip/irq-gic-v3-its.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/irqchip/irq-gic-v3-its.c b/drivers/irqchip/irq-gic-v3-its.c
index 1994541eaef8..034ece9ac47c 100644
--- a/drivers/irqchip/irq-gic-v3-its.c
+++ b/drivers/irqchip/irq-gic-v3-its.c
@@ -4727,7 +4727,8 @@ static bool __maybe_unused its_enable_rk3588001(void *data)
 {
 	struct its_node *its = data;
 
-	if (!of_machine_is_compatible("rockchip,rk3588"))
+	if (!of_machine_is_compatible("rockchip,rk3588") &&
+	    !of_machine_is_compatible("rockchip,rk3588s"))
 		return false;
 
 	its->flags |= ITS_FLAGS_FORCE_NON_SHAREABLE;
